options/rtld/generic/linker.hpp

```diff
@@ -86,6 +89,9 @@ struct ObjectRepository {
 
 	frg::hash_map<frg::string_view, SharedObject *,
 			frg::hash<frg::string_view>, MemoryAllocator> _nameMap;
+
+	// Used for destructing the objects, stores all the objects in the order they are initialized.
+	frg::vector<SharedObject *, MemoryAllocator> _destructQueue;
```

Comment:
Nitpick but you could use a ``frg::stack`` here since destruction happens in reverse order anyway.